tests/components/login/login_controller/login_controller.test.jsx

```diff
@@ -98,4 +98,23 @@ describe('components/login/LoginController', () => {
 
         expect(wrapper).toMatchSnapshot();
     });
+
+    it('should discard session expiry notification on failed sign in', () => {
+        const props = {
+            ...baseProps,
+            initializing: false,
+            location: {
+                search: '?extra=' + Constants.SIGNIN_CHANGE,
+            },
+        };
+
+        LocalStorageStore.setWasLoggedIn(true);
+        const wrapper = shallowWithIntl(
+            <LoginController {...props}/>
+        ).dive();
+
+        const e = {preventDefault: jest.fn()};
+        wrapper.instance().preSubmit(e);
+        expect(wrapper).toMatchSnapshot();
```

Comment:
Maybe instead of using a snapshot or the LocalStorageStore here, we could use the state directly?

```javascript
const wrapper = shallowWithIntl(<LoginController {...props}/>);

wrapper.setState({sessionExpired: true});

const e = {preventDefault: jest.fn()};
wrapper.instance().preSubmit(e);

expect(wrapper.state('sessionExpired')).toBe(false);
```

Reply:
Brilliant! Done.